pandas/core/array_algos/putmask.py

```diff
@@ -73,9 +72,6 @@ def putmask_without_repeat(
     mask : np.ndarray[bool]
     new : Any
     """
-    if np_version_under1p21:
-        new = setitem_datetimelike_compat(values, mask.sum(), new)
```

Comment:
i suspect setitem_datetimelike_compat can be deleted entirely

Reply:
There are two other usages:

https://github.com/pandas-dev/pandas/blob/249d93e4abc59639983eb3e8fccac8382592d457/pandas/core/indexes/base.py#L5327

https://github.com/pandas-dev/pandas/blob/249d93e4abc59639983eb3e8fccac8382592d457/pandas/core/internals/blocks.py#L1059

Reply:
my hope was that with the bumped version those would be unnecessary too.  i guess not?

Reply:
Removing it creates many test failures. I'll leave it for now.

Reply:
Darn.  Thanks for checking